### PR TITLE
Update modal

### PR DESCRIPTION
**Pull request by surekhaw**
## Description
pull current product info into modal to compare with related products. Dynamically render rows with features to compare. If there is a common feature, the value for the current product and compared product will appear on the same line (if the value is "true", it will appear as a check mark). If they do not have features in common, these feature rows will also appear, with a value in the column of the product that has it, and blank for the product that doesn't. False values will appear blank. Scroll feature has also been added in case there are two many features to fit within the modal window.

## How to test
manually. name of both current product and related product will appear in modal when clicking on star button of related product card. features will appear in the center column. Values of the respective product will appear in the respect column as a string or a check mark.

## Notes
will write tests in separate ticket. also added some styling. did not see in business doc where it specified how to close modal, and the visual design does not have a button. For now, click on the modal to close it once open. the current related products and current product do not have any boolean values, so will have to write tests in separate ticket to check for this feature, as well as scroll feature.

## Issue tracking
https://trello.com/c/5ovhTMcK/110-m-pull-current-product-info-from-store-into-modal
https://trello.com/c/SXwRSD4M/114-s-update-close-button-on-modal
https://trello.com/c/qtO1JYKp/126-m-add-product-specific-characteristics-to-modal

### CODE REVIEW

File: client/src/components/relatedProducts/Modal.jsx [hunk 1]
```diff
@@ -1,65 +1,63 @@
...
 // shows onClick star icon
 const Modal = (props) => {
-  const { comparedProduct } = props;
+  const { comparedProduct, currentProduct } = props;
+  const features = {};
+  if (comparedProduct.product !== undefined && comparedProduct.product.features !== undefined) {
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < currentProduct.features.length; i++) {
+      if (currentProduct.features[i].value === true) {
+        currentProduct.features[i].value = <FontAwesomeIcon icon={'faCheck'} />;
```
Comment (tm2838) [inline client/src/components/relatedProducts/Modal.jsx:13]: lol have we added this icon to our icon library? looks like it's not showing up for me 👀

**Review by tm2838 — APPROVED** [review]
Modal looks good! only blocker is that the check mark icon not showing up

File: client/src/components/relatedProducts/Modal.jsx (hunk 1 above)
Comment (surekhaw) [reply client/src/components/relatedProducts/Modal.jsx:13]: check mark is only supposed to show if one of the values is the boolean true, which none of the products have that are currently rendering on the page. However, I will also add to the library in the "common" folder. I thought the "import fontawesome from @fortawesome..." as opposed to importing the file meant that adding to the library was not necessary. Sorry!

Comment (surekhaw) [reply client/src/components/relatedProducts/Modal.jsx:13]: added check mark to library

**Pull request merged**